packages/nimble-components/src/wafer-map/index.ts

```diff
@@ -244,14 +267,64 @@ export class WaferMap extends FoundationElement {
                 this.dataManager.updateDiesRenderInfo();
                 this.renderer.updateSortedDiesAndDrawWafer();
             } else if (this.waferMapUpdateTracker.requiresDrawnWaferUpdate) {
-                this.renderer.drawWafer();
+                await this.renderer.drawWafer();
             }
             this.zoomHandler.connect();
```

Comment:
While I understand the change you made, I'm still curious about how the `zoomHandler` actually operates. Why is it necessary to recreate the `ZoomBehavior` for all updates tracked by `requiresEventUpdate`? That tracked event seems to be a catch-all for most other tracked state changes (for some reason), including things that don't seem related to state that would require a re-creation of the `ZoomBehavior`.

Reply:
Indeed it is not needed to connect and disconnect at every update, only when the width/height of the canvas is needed, I will update the code accordingly